### DISCUSSION TITLE
Is `https://mjml.io/try-it-live` open sourced

### DISCUSSION THREAD

**Discussion by freizl** [mjmlio/mjml#2701] [General]
I was curious where does those validation rule exists?
e.g. 
```
mj-text cannot be used inside mj-section, only inside: mj-attributes, mj-column, mj-hero
```

I looked at the `mjml-validator` package, which looks like core but no actual rules live there.
(Assuming the `registerRule` will be called some where to actually create validation rules)

In addition, even though there is validation errors, the page still render properly.

Thanks!

**Comment by iRyusa** [comment]
Dependencies are defined here https://github.com/mjmlio/mjml/blob/170db3f09b0e2d49e95167314134ad4377b29588/packages/mjml-preset-core/src/dependencies.js and most rules rely on a pseudo-typed attributes definition in  components
